src/main/java/org/broadinstitute/hellbender/tools/spark/sv/CollectInsertSizeMetricsSpark.java

```diff
@@ -0,0 +1,231 @@
+              shortName = "nPP",
+              fullName = "useNonProperlyPairedReads",
+              optional = true)
+    public boolean useNonProperlyPairedReads = false;
+
+    @Argument(doc = "If set to true, include unmapped reads as well.",
+              shortName = "U",
+              fullName = "useUnmappedReads",
+              optional = true)
+    public boolean useUnmappedReads = false;
+
+    @Argument(doc = "If set to true, include reads whose mate is unmapped as well.",
+              shortName = "Um",
+              fullName = "useMateUnmappedReads",
+              optional = true)
+    public boolean useMateUnmappedReads = false;
+
+    @Argument(doc = "If set to true, include duplicated reads as well.",
+              shortName = "Dup",
+              fullName = "useDupReads",
+              optional = true)
+    public boolean useDupReads = false;
+
+    @Argument(doc = "If set to true, include secondary alignments.",
+              shortName = "S",
+              fullName = "useSecondaryAlignments",
+              optional = true)
+    public boolean useSecondaryAlignments = false;
+
+    @Argument(doc = "If set to true, include supplementary alignments.",
+              shortName = "SS",
+              fullName = "useSupplementaryAlignments",
+              optional = true)
+    public boolean useSupplementaryAlignments = false;
+
+    @Argument(doc = "If set non-zero value, only include reads passing certain mapping quality threshold. " +
+                    "If set to zero, reads with zero mapping quality will be included in calculating metrics.",
+              shortName = "Q",
+              fullName = "MQPassingThreshold",
+              optional = true)
+    public int MQPassingThreshold = 0;
+
+//    TODO
+//    @Argument(doc = "If set to non-zero value, only include reads with mate passing certain mapping quality threshold.",
+//              shortName = "Qm",
+//              fullName = "mateMQPassingThreshold",
+//              optional = true)
+//    public int mateMQPassingThreshold = 0;
+
+//    @Argument(doc="The level(s) at which to accumulate metrics. Possible values are {SAMPLE, LIBRARY, READ GROUP}. " +
+//                  "Default: All reads.",
+//              shortName="LEVEL",
+//              fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME,
+//              optional = false)
+//    public Set<MetricAccumulationLevel> METRIC_ACCUMULATION_LEVEL = EnumSet.of(MetricAccumulationLevel.ALL_READS);
+
+    /**
+     * Which end of a read pair to use for collecting insert size metrics.
+     * TODO: For truncated genomic regions of putative SV breakpoints, not all reads have both ends land in the region,
+     *       so third case is possible: will use either end when only one end is available in the region specified,
+     *       and only first end if both are available
+     */
+    public enum whichEndToUse{
```

Comment:
according to java style guide https://google.github.io/styleguide/javaguide.html#s4.8.1-enum-classes enum classes should be named `LikeThis` and the constants (enum objects) should be named `LIKE_THIS`